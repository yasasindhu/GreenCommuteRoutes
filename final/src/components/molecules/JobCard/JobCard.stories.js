import React from "react";

import JobCard from "./JobCard";

import LocalTaxiOutlinedIcon from '@material-ui/icons/LocalTaxiOutlined';
import TrainOutlinedIcon from '@material-ui/icons/TrainOutlined';
import DirectionsBusOutlinedIcon from '@material-ui/icons/DirectionsBusOutlined';
import MotorcycleOutlinedIcon from '@material-ui/icons/MotorcycleOutlined';

export default {
    title: "molecules/Job Card Expanded",
    component: JobCard,
};

const Template = (args) => <JobCard {...args} />;

export const PrimaryJobCard = Template.bind({});


PrimaryJobCard.args = {
    job:{
        logo : "https://cdn.zeplin.io/5fa28fcbff79c58958e4e1a6/assets/4a211710-3383-49de-9edd-ef48e3af27d1-2x.png",
        roleName : 'User Experience Designer',
        companyName: 'Hp',
        location : 'Hyderabad, Telangana India',
        cons : [<LocalTaxiOutlinedIcon />, <TrainOutlinedIcon /> , <DirectionsBusOutlinedIcon /> , <MotorcycleOutlinedIcon />]
    }
};