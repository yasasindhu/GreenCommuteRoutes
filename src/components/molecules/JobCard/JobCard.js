
import React from "react";
import { makeStyles, ThemeProvider } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";

import MyImage from "../../atoms/MyImage/MyImage";
import MoreHorizIcon from "@material-ui/icons/MoreHoriz";

import customtheme from '../../theme.jsx';
import Typography from '@material-ui/core/Typography';

import LocalTaxiOutlinedIcon from '@material-ui/icons/LocalTaxiOutlined';
import TrainOutlinedIcon from '@material-ui/icons/TrainOutlined';
import DirectionsBusOutlinedIcon from '@material-ui/icons/DirectionsBusOutlined';
import MotorcycleOutlinedIcon from '@material-ui/icons/MotorcycleOutlined';
import { useDispatch, useSelector } from "react-redux";
import { showJob } from "../../../store/actions/actions";
import { useHistory } from 'react-router-dom';


const useStyles = makeStyles({
	root: {
		width: '330px',
		height: '308px',
		// margin: "20px 0px 20px 60px",
		marginTop: "20px",
		marginRight: "15px",
		marginLeft: "15px",
		'&:hover': {
			borderColor: '#0365f2',
		}
	},
	root2: {
		maxWidth: '690px',
		maxHeight: '138px',
		padding: '1.5%',
		borderRadius: '10px'
	},
	logoPadding: {
		paddingLeft: '7%',
		paddingTop: '7%',
		paddingRight: '3%',
		paddingBottom: '3%'
	},
	header: {
		display: "flex",
		justifyContent: "space-between",
		alignItems: "center",
		objectFit: 'contain',
	},
	spacing: {
		color: '#5f7381',
		width: '52px',
		flexGrow: '0',
		display: 'flex',
		flexDirection: 'row',
		justifyContent: 'flex-spacearound',
		paddingRight: "7%",
		alignItems: 'center',
		gap: '10px',
		padding: '0',
		paddingBottom: '50px'
	},
	inCol: {
		display: "flex",
		flexDirection: "column",
		justifyContent: "space-between",
	},
	iconGrp: {
		display: 'inline-flex',
		flexDirection: 'row',
		justifyContent: 'flexStart',
		alignItems: 'center',
		paddingTop: '3%',
		// gap : '8px'
	},
	roleTypoHeader: {
		fontSize: '18px',
		fontWeight: '600',
		color: '#324552',
		lineHeight: 1.33,
		fontStretch: 'normal',
		fontStyle: 'normal',
		letterSpacing: '0.2px'
	},
	companyLocationTypoHeader: {
		fontSize: '16px',
		fontWeight: 'normal',
		color: '#5f7381',
		lineHeight: 1.57,
		fontStretch: 'normal',
		fontStyle: 'normal',
		letterSpacing: '0.1px',
		paddingTop: '2%'
	},

	padding: {
		paddingTop: '0',
		lineHeight: 1.33
	},
	jobInfoPadding: {
		paddingLeft: '20px'
	},
	moreIconPadding: {
		color: '#5f7381',
	}

});

export default function JobCard({
	job
}) {
	const logo = job.logo
	const roleName = job.roleName
	const companyName = job.companyName
	const location = job.location
	const classes = useStyles();
	const icons = [[<LocalTaxiOutlinedIcon />, <TrainOutlinedIcon />, <DirectionsBusOutlinedIcon />, <MotorcycleOutlinedIcon />]]
	const dispatch = useDispatch();
	const stateForRenderingJobDescriptioninRight = useSelector((state => state.jobsDescription))
	const history = useHistory()
	const handleChangeForCard = () => {
		dispatch(showJob(job)),
			history.push('/findJobs')
	}
	return (
		<ThemeProvider theme={customtheme}>
			<Card className={classes.root} variant="outlined" data-testid="cardinjobcard" onClick={handleChangeForCard}>
				<div className={classes.header} data-testid="upperdivinjobcard">
					<MyImage
						className={classes.logoPadding}
						alt={logo}
						src={logo}
						width="80px"
						height="80px"
					/>
					<div className={` ${classes.spacing}`} data-testid="innerdivinjobcard">
						<Typography variant="h1" data-testid="2dtypographyinjobcard">
							2d
						</Typography>
						<MoreHorizIcon
							className={classes.moreIconPadding}
							fontSize="small"
							data-testid="morehorizoninjobcard"
						/>
					</div>
				</div>
				<CardContent data-testid="cardcontentinjobcard">
					<Typography gutterBottom variant="h3" data-testid="rolenameinjobcard">
						{roleName}
					</Typography>

					<Typography variant="h6" data-testid="companynameinjobcard">
						{companyName}
					</Typography>
					<Typography variant="h6" data-testid="locationinjobcard">
						{location}
					</Typography>

					<Typography gutterBottom variant="h5" data-testid="commuterouteinjobcard">
						Commute routes available:
					</Typography>
					<div className={classes.iconGrp} data-testid="iconnameinjobcard">
						{icons && icons.map(iconName => (
							<span style={{ color: '#9bbdcb' }} >{iconName}</span>
						))}

					</div>
				</CardContent>
			</Card>
		</ThemeProvider>
	);
}