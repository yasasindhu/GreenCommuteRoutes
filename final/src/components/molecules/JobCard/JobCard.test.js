import { render } from '@testing-library/react';
import JobCard from './JobCard';

import jobCardDetails from '../../Organism/RenderJobCard/jobCardDetails'

describe('Typography Rendered', () => {
  test('check the Descption Card title', () => {
      const { getByTestId } = render(<JobCard job = {jobCardDetails[0]}/>);
      expect(getByTestId('cardinjobcard')).toBeTruthy();
  });
});