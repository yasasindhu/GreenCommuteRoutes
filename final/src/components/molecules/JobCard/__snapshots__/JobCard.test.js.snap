// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Rendered check the Descption Card title 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      data-testid="cardinjobcard"
    >
      <div
        class="makeStyles-header-4"
        data-testid="upperdivinjobcard"
      >
        <img
          alt="ola.png"
          class="makeStyles-logoPadding-3"
          data-testid="image"
          height="80px"
          src="ola.png"
          width="80px"
        />
        <div
          class=" makeStyles-spacing-5"
          data-testid="innerdivinjobcard"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1"
            data-testid="2dtypographyinjobcard"
          >
            2d
          </h1>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-moreIconPadding-12 MuiSvgIcon-fontSizeSmall"
            data-testid="morehorizoninjobcard"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
        data-testid="cardcontentinjobcard"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          data-testid="rolenameinjobcard"
        >
          Manager
        </h3>
        <h6
          class="MuiTypography-root MuiTypography-h6"
          data-testid="companynameinjobcard"
        >
          Zemoso
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6"
          data-testid="locationinjobcard"
        >
          Hyd
        </h6>
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          data-testid="commuterouteinjobcard"
        >
          Commute routes available:
        </h5>
        <div
          class="makeStyles-iconGrp-7"
          data-testid="iconnameinjobcard"
        >
          <span
            style="color: rgb(155, 189, 203);"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5H15V3H9v2H6.5c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.85 7h10.29l1.04 3H5.81l1.04-3zM19 17H5v-4.66l.12-.34h13.77l.11.34V17z"
              />
              <circle
                cx="7.5"
                cy="14.5"
                r="1.5"
              />
              <circle
                cx="16.5"
                cy="14.5"
                r="1.5"
              />
            </svg>
          </span>
          <span
            style="color: rgb(155, 189, 203);"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="8.5"
                cy="14.5"
                r="1.5"
              />
              <circle
                cx="15.5"
                cy="14.5"
                r="1.5"
              />
              <path
                d="M12 2c-4 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h2l2-2h4l2 2h2v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-4-4-8-4zm0 2c3.51 0 4.96.48 5.57 1H6.43c.61-.52 2.06-1 5.57-1zM6 7h5v3H6V7zm12 8.5c0 .83-.67 1.5-1.5 1.5h-9c-.83 0-1.5-.67-1.5-1.5V12h12v3.5zm0-5.5h-5V7h5v3z"
              />
            </svg>
          </span>
          <span
            style="color: rgb(155, 189, 203);"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2c-4.42 0-8 .5-8 4v10c0 .88.39 1.67 1 2.22V20c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h8v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.78c.61-.55 1-1.34 1-2.22V6c0-3.5-3.58-4-8-4zm5.66 2.99H6.34C6.89 4.46 8.31 4 12 4s5.11.46 5.66.99zm.34 2V10H6V6.99h12zm-.34 9.74l-.29.27H6.63l-.29-.27C6.21 16.62 6 16.37 6 16v-4h12v4c0 .37-.21.62-.34.73z"
              />
              <circle
                cx="8.5"
                cy="14.5"
                r="1.5"
              />
              <circle
                cx="15.5"
                cy="14.5"
                r="1.5"
              />
            </svg>
          </span>
          <span
            style="color: rgb(155, 189, 203);"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.72 11l-2 2h-.77l-.25-.69c-.18-.48-.42-.92-.72-1.31h3.74m2.69-6H11v2h3.59l2 2H5c-2.8 0-5 2.2-5 5s2.2 5 5 5c2.46 0 4.45-1.69 4.9-4h1.65l2.77-2.77c-.21.54-.32 1.14-.32 1.77 0 2.8 2.2 5 5 5s5-2.2 5-5c0-2.65-1.97-4.77-4.56-4.97L15.41 5zM19 17c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zM5 17c-1.63 0-3-1.37-3-3s1.37-3 3-3c1.28 0 2.4.85 2.82 2H5v2h2.82C7.4 16.15 6.28 17 5 17z"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
